src/ngraph/op/non_max_suppression.hpp

```diff
@@ -0,0 +1,93 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v1
+        {
+            /// \brief Elementwise addition operation.
+            ///
+            class NGRAPH_API NonMaxSuppression : public Op
+            {
+            public:
+                enum class BoxEncodingType
+                {
+                    CORNER,
+                    CENTER
+                };
+
+                static constexpr NodeTypeInfo type_info{"NonMaxSuppression", 1};
+                const NodeTypeInfo& get_type_info() const override { return type_info; }
+                NonMaxSuppression() = default;
+
+                /// \brief Constructs a NonMaxSuppression operation.
+                ///
+                /// \param boxes Output that produces a tensor with box coordinates.
+                /// \param scores Output that produces ta tensor
+                /// \param max_output_boxes_per_class Auto broadcast specification
+                /// \param iou_threshold Auto broadcast specification
+                /// \param score_threshold Auto broadcast specification
+                /// \param box_encoding Auto broadcast specification
+                /// \param sort_result_descending Auto broadcast specification
+                NonMaxSuppression(const Output<Node>& boxes,
+                                  const Output<Node>& scores,
+                                  const Output<Node>& max_output_boxes_per_class,
+                                  const Output<Node>& iou_threshold,
+                                  const Output<Node>& score_threshold,
+                                  const BoxEncodingType box_encoding = BoxEncodingType::CORNER,
+                                  const bool sort_result_descending = true);
+
+                /// \brief Constructs a NonMaxSuppression operation with default values for the last
+                ///        3 inputs
+                ///
+                /// \param boxes Output that produces a tensor with box coordinates.
+                /// \param scores Output that produces ta tensor
+                /// \param box_encoding Auto broadcast specification
+                /// \param sort_result_descending Auto broadcast specification
+                NonMaxSuppression(const Output<Node>& boxes,
+                                  const Output<Node>& scores,
+                                  const BoxEncodingType box_encoding = BoxEncodingType::CORNER,
```

Comment:
In the spec we have 3 required inputs: boxes, scores and score_threshold.

Reply:
The current version of the spec excplicitly mentions that `max_output_boxes_per_class` is optional. Can you double check?

Reply:
Yes, sorry, my fault.  score_threshold - Optional with default value 0